test/e2e/network/network_policy.go

```diff
@@ -1450,9 +1453,15 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
 				framework.ExpectNoError(err, "Error occurred while getting pod status.")
 			}
 
-			podServerAllowCIDR := fmt.Sprintf("%s/24", podServerStatus.Status.PodIP)
+			allowMask := 24
+			exceptMask := 32
+			if utilnet.IsIPv6String(podServerStatus.Status.PodIP) {
+				allowMask = 64
+				exceptMask = 128
+			}
+			podServerAllowCIDR := fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, allowMask)
```

Comment:
oops, we're testing undefined behavior anyway; we're saying to allow to, eg, "192.168.1.5/24" when we should be saying to allow to "192.168.1.**0**/24".

You need to do:

    _, podServerAllowSubnet, err := net.ParseCIDR(fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, allowMask))
    framework.ExpectNoError(err, "could not parse allow subnet")
    podServerAllowCIDR := podServerAllowSubnet.String()

likewise in the test below

Reply:
yeah, strictly speaking that's correct, just curiosity, do you think that app should "normalize it" and use it or reject it?

Reply:
There's no way for us to know whether the user actually meant "192.168.1.5/**32**" or "192.168.1.**0**/24"; either mistake is possible. So we can't "normalize it". Ideally we would have rejected this at validation time.